templates/gitlab.rb.j2

```diff
@@ -49,10 +49,16 @@ nginx['listen_https'] = "{{ gitlab_nginx_listen_https }}"
 gitlab_rails['smtp_enable'] = {{ gitlab_smtp_enable }}
 gitlab_rails['smtp_address'] = '{{ gitlab_smtp_address }}'
 gitlab_rails['smtp_port'] = {{ gitlab_smtp_port }}
+{% if gitlab_smtp_authentication == 'false' %}
+gitlab_rails['smtp_domain'] = '{{ gitlab_smtp_domain }}'
```

Comment:
It looks like smtp_domain and smtp_authentication are used in either case... so maybe we can just have:

```
{% if gitlab_smtp_authentication == 'false' %}
gitlab_rails['smtp_user_name'] = '{{ gitlab_smtp_user_name }}'
gitlab_rails['smtp_password'] = '{{ gitlab_smtp_password }}'
{% endif %}
gitlab_rails['smtp_domain'] = '{{ gitlab_smtp_domain }}'
gitlab_rails['smtp_authentication'] = '{{ gitlab_smtp_authentication }}'
```

Reply:
This feature is needed. What can I do else?